src/reason_pprint_ast.ml

```diff
@@ -697,6 +699,16 @@ let rules = [
   [
     (TokenPrecedence, (fun s -> (Nonassoc, s = "=>")));
   ];
+  [
+    (TokenPrecedence, (fun s ->
+      let is_infix_op = Hashtbl.mem infix_operators s in
```

Comment:
The puzzle is now we have `%left VARIABLEINFIXOP` in the parser, saying this is a left-assoc operator.

With your proposal we have to somehow tell the parser to parse it as %right VARIABLEINFIXOP. Not sure how to do it with Menhir (our parser generator).